fdbserver/storageserver.actor.cpp

```diff
@@ -4819,26 +4835,21 @@ ACTOR Future<GetMappedKeyValuesReply> mapKeyValues(StorageServer* data,
 	state std::vector<MappedKeyValueRef> kvms(k);
 	state std::vector<Future<Void>> subqueries;
 	state int offset = 0;
+	state int cnt = 0;
 	if (pOriginalReq->options.present() && pOriginalReq->options.get().debugID.present())
 		g_traceBatch.addEvent("TransactionDebug",
 		                      pOriginalReq->options.get().debugID.get().first(),
 		                      "storageserver.mapKeyValues.BeforeLoop");
-	for (; offset < sz; offset += SERVER_KNOBS->MAX_PARALLEL_QUICK_GET_VALUE) {
+
+	for (; offset<sz&& * remainingLimitBytes> 0; offset += SERVER_KNOBS->MAX_PARALLEL_QUICK_GET_VALUE) {
```

Comment:
this is a great question, basically to clarify on the case where we have unlimited byteLimit. 

* it is handled [here](https://github.com/apple/foundationdb/blob/main/fdbclient/NativeAPI.actor.cpp#L4439) then [here](https://github.com/apple/foundationdb/blob/main/fdbclient/NativeAPI.actor.cpp#L3948-L3949), and set to `CLIENT_KNOBS->REPLY_BYTE_LIMIT`. So if rowLimit wants more, there is nothing we can do anyways.


However I found a bug -- we actually allow either rowLimit or byteLimit to be empty, rather than only enforce rowLimit [code](https://github.com/apple/foundationdb/blob/ca33fe1f1eefd2afa0b50d281cc3699454c18c8c/bindings/c/fdb_c.cpp#L722-L724)